vault/token_store.go

```diff
@@ -1123,27 +1124,31 @@ func (ts *TokenStore) revokeSalted(ctx context.Context, saltedID string) (ret er
 		return err
 	}
 
-	// If we are returning an error, mark the entry with -3 to indicate
-	// failed revocation. This way we don't try to clean up during active
-	// revocation (-2).
 	defer func() {
-		if ret != nil {
+		if deleteEntry || ret != nil {
```

Comment:
This is still potentially racy (if `ts.view.Delete` returns an error it doesn't grab a lock). Gotta think through this a bit more.

Reply:
Actually this should be alright, since the lock will be held whenever `deleteEntry` is true.

Reply:
You could just shortcut this -- if deleteEntry is not true, and ret is nil, just return.